### DISCUSSION TITLE
How do VIMP values compare to %IncMSE importance from randomforest package

### DISCUSSION THREAD

**Discussion by mc19952004** [kogalur/randomForestSRC#155] [Q&A]
Hi
 
I hope you are well.
 
I have recently been using your package for multivariate random forest modelling, and I had a question relating to the variable importance calculations. I was wondering how the values relate to the importance values obtained from a standard random forest model from the ‘randomforest’, where the importance values are given as % increase in MSE and the Gini Index. I understand that the corresponding calculation approach similar to the ‘randomforest’ package would be the importance= “permute” within your package, but wanted to know what the output values for importance in your package are.
 
Thanks
 
Michael

**Comment by ishwaran** [comment]
The first thing to note is that the importance (VIMP) values are given for each outcome.  So if you have a mixture of continuous and factor outcomes, you will get VIMP values for regression and VIMP values for classification.  In the first case, VIMP is measured by mean-squared-error and in the second case, by misclassification error.  In order to extract VIMP for all outcomes, you should consider using the function `get.mv.vimp`.  An illustration of the function is given in the `nutrigenomic `help file [https://luminwin.github.io/randomForestSRC/reference/nutrigenomic.html](url)

The default VIMP used by the package is not permutation importance (which you can always obtain using the option `importance='permute'`) but actually anti importance---the reason for this, is that anti-VIMP is much faster than permutation-VIMP.

Thus to summarize: what is returned is mean-squared error VIMP for continuous outcome values and misclassification error VIMP for the factor outcome values.  The default technique for calculating VIMP is anti, however you can always request the standard permutation importance if you want.
